drivers/hmis_external_apis/app/models/hmis_external_apis/ac_hmis/importers/loaders/deferred_project_unit_occupancy_loader.rb

```diff
@@ -0,0 +1,45 @@
+###
+# Copyright 2016 - 2023 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# matriculation to new platform
+module HmisExternalApis::AcHmis::Importers::Loaders
+  class DeferredProjectUnitOccupancyLoader < BaseLoader
+    def initialize(tracker:, clobber:)
+      super(reader: nil, tracker: tracker, clobber: clobber)
+    end
+
+    def runnable?
+      super && tracker.assignments.any?
+    end
+
+    def perform
+      enrollments = Hmis::Hud::Enrollment.where(data_source: data_source)
+      scoped_records = model_class.where(enrollment_id: enrollments.select(:id))
+      # destroy all existing records
+      scoped_records.destroy_all if clobber
+      ar_import(model_class, build_records, recursive: true)
+    end
+
+    protected
+
+    def build_records
+      rows.map do |row|
+        unit_id, enrollment_id, start_date = row.fetch_values(:unit_id, :enrollment_id, :start_date)
+        record = model_class.new(unit_id: unit_id, enrollment_id: enrollment_id)
+        record.build_occupancy_period(start_date: start_date || today, user_id: system_user_id)
```

Comment:
Welp, `user_id` on this table should be the primary key for the `Hmis::User` (`users` table). I'll look for other usages of `system_user_id` and check them.

```suggestion
        record.build_occupancy_period(start_date: start_date || today, user_id: Hmis::User.system_user.id)
```


Its confusing, but the rule is:
* If it's a "hud" table, the table name is camelcase and so are the column names. it should have a column `UserID`. In this case, `UserID` maps to the `UserID` column on the `Users` table.
* If it's a "non-hud" table, the table name is snake-cased and so are the column names. it may have a column `user_id`. In this case, `user_id` maps to the `id` column on the `users` table.

This has tripped me up sooo many times, I wish it weren't this way, but it is. cc @eanders we don't have this documented anywhere, do we? we should, def a quirk when onboarding to the codebase (And beyond)

Reply:
Yeah, not documented that I know of.  Leave it to HUD to slip in the Users table a few years ago.  I don't have a better solution, not really sure where to document it, other than maybe the user.rb files.

Reply:
Okay. yeah I was thinking something on the "HUD tables" vs "non HUD tables" more broadly in the readme somewhere. To do before onboarding the next contributor 🙂

Reply:
Great catch! I missed that we were using the pk here. Thanks for auditing the other uses of system_user_id. It's too bad we can't have a FK back to user here, that would be a nice safety check.